src/Illuminate/Database/DatabaseTransactionsManager.php

```diff
@@ -54,30 +67,28 @@ public function rollback($connection, $level)
         $this->transactions = $this->transactions->reject(
             fn ($transaction) => $transaction->connection == $connection && $transaction->level > $level
         )->values();
-
-        if ($this->transactions->isEmpty()) {
-            $this->callbacksShouldIgnore = null;
-        }
     }
 
     /**
      * Commit the active database transaction.
      *
      * @param  string  $connection
+     * @param  int  $level
      * @return void
      */
-    public function commit($connection)
+    public function commit($connection, $level)
```

Comment:
Would it be possible if we set `$level = 1` here since this would be the normal default value except during tests?

Reply:
Hmm, I don't see a value in adding the default value here. The method is only called in from within the `ManagesTransactions` trait inside the framework. Are you worried about other libs calling this method manually? Adding the default value might give the wrong impression that this is optional. If we add a default value of `1` and people call this method without passing the transaction level, we'll always execute the callbacks no matter which transaction level we're currently in... so I'd be careful there. 🤔

Reply:
`db.transaction` binding has been available since Laravel 9, so someone might already be extended or overridden to cater to other use cases such as databases without transactions such as Mongodb, etc. 

Also while it's only called from the class the visibility of the method is still `public`, wouldn't mind much if it was `protected` or `private`.

Reply:
Added